scipy/optimize/tests/test_lsq_linear.py

```diff
@@ -76,6 +77,18 @@ def test_dense_bounds(self):
             assert_allclose(res.x, np.array([0.005236663400791, -4]))
             assert_allclose(res.unbounded_sol[0], unbounded_sol)
 
+    def test_bounds_variants(self):
+        x = np.array([1, 3])
+        A = self.rnd.uniform(size=(2, 2))
+        b = A@x
+        lb = np.array([1, 1])
+        ub = np.array([2, 2])
+        bounds_old = (lb, ub)
+        bounds_new = Bounds(lb, ub)
+        res_old = lsq_linear(A, b, bounds_old)
+        res_new = lsq_linear(A, b, bounds_new)
+        assert_allclose(res_old.x, res_new.x)
```

Comment:
To emphasize that this is a problem for which `bounds` have an effect on the solution:
```suggestion
        assert_allclose(res_old.x, res_new.x)
        assert not np.allclose(res_new.x, res_new.unbounded_sol[0])
```
(please check locally that this is correct)

Reply:
done